Content.Server/Power/EntitySystems/ExtensionCableSystem.cs

```diff
@@ -248,6 +249,7 @@ private bool TryFindAvailableProvider(EntityUid owner, float range, [NotNullWhen
             var coordinates = xform.Coordinates;
             var nearbyEntities = grid.GetCellsInSquareArea(coordinates, (int) Math.Ceiling(range / grid.TileSize));
 
+            var candidates = new List<(float Distance, ExtensionCableProviderComponent Provider)>();
```

Comment:
I just realized there's a way better way to do this whole thing that doesn't need a list at all:

You can just store the closest provider _so far_, and it's distance. Then in the loop you can just check if the new component is closer than the last-closest.

Avoids needing to allocate a list and sort it.

Reply:
Good catch! I missed this initially as I got over-eager to try out LINQ :-)